crate_universe/src/cli/splice.rs

```diff
@@ -56,18 +50,14 @@ pub fn splice(opt: SpliceOptions) -> Result<()> {
     // Load the all config files required for splicing a workspace
     let splicing_manifest = SplicingManifest::try_from_path(&opt.splicing_manifest)?;
 
-    // Determine the splicing workspace
-    let temp_dir;
-    let splicing_dir = match &opt.workspace_dir {
-        Some(dir) => dir.clone(),
-        None => {
-            temp_dir = tempfile::tempdir().context("Failed to generate temporary directory")?;
-            temp_dir.as_ref().to_path_buf()
-        }
-    };
-
-    // Generate a splicer for creating a Cargo workspace manifest
-    let splicer = Splicer::new(splicing_dir, splicing_manifest)?;
+    // Generate a splicer for creating a Cargo workspace manifest. We splice
+    // directly into the output_dir because we may need to subsequently run
+    // "cargo tree ..." here and will need everything (Cargo.toml, all symlinks)
+    // present.
+    let splicer = Splicer::new(
+        opt.output_dir.clone(), /*splicing_dir*/
```

Comment:
Why do we need to run subsequent cargo tree commands?

Additionally, are we no longer splicing in a temp directory? There were issues that came from splicing out of temp directory.

Reply:
To generate the feature map -- as implemented currently this is done as part of the "generate" command instead of the "splice" command.

I suppose instead I could generate the feature map as part of the splicing step and save it into a new file adjacent to the metadata file and `Cargo.lock`?

Reply:
You can save it directly to the `Cargo.toml` file in the metadata section. There's already a class which populates this and is consumed downstream of the splicing step:
https://github.com/bazelbuild/rules_rust/blob/main/crate_universe/src/splicing.rs#L151-L153

Ideally, all calls to `cargo` would be contained to the splicing step and everything else is deterministic computations and renderings of the data produced at that step.

Reply:
Ah, very cool. I'll update to use this approach and circle back with you. Thank you!!

Reply:
Done -- the feature map is now written into `WorkspaceMetadata` during the splicing phase. Thanks!